tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -462,9 +462,12 @@ public:
    {
       auto rColl = fColls[0];
       for (unsigned int i = 1; i < fColls.size(); ++i) {
-         auto &coll = fColls[i];
-         for (auto &&v : *coll) {
-            FillColl(v, *rColl);
+         const auto &coll = fColls[i];
+         const auto begin = coll->begin();
+         // Use an explicit loop here to prevent compiler warnings introduced by
+         // clang's range-based loop analysis and vector<bool> references.
+         for (std::size_t j = 0; j < coll->size(); j++) {
+            FillColl(*std::next(begin, j), *rColl);
```

Comment:
I think it's more performant to increment the iterator in the for loop and avoid calls to `next` and `size`:

```c++
const auto end = coll->end(); 
for (auto it = coll->begin(); it != end; ++it)
   FillColl(*it, ...);
```

Reply:
also `cbegin` and `cend` would be (admittedly very slightly) better

Reply:
Absolutely! Thanks :)